### PR TITLE
feat: add SemanticTextSplitter using EmbeddingModel cosine similarity

### LINKED ISSUE #5464: Add native semantic text chunking support
**Expected Behavior**

Spring AI should provide a native SemanticTextSplitter that uses an EmbeddingModel to split text based on semantic similarity between sentences, producing higher-quality chunks for RAG pipelines.

SemanticTextSplitter splitter = SemanticTextSplitter.builder()
    .embeddingModel(embeddingModel)
    .similarityThreshold(0.5)
    .maxChunkSize(1000)
    .build();

List<Document> chunks = splitter.split(documents);

**The algorithm:**
1-Split text into sentences
2-Compute embeddings for each sentence (or sliding window of sentences)
3-Calculate cosine similarity between consecutive sentence embeddings
4-Split at points where similarity drops below the configured threshold
5-This produces variable-sized chunks that respect natural semantic boundaries of the text.

**Current Behavior**
The only native text splitter is TokenTextSplitter, which splits based on a fixed token count. This can break semantically related content across chunk boundaries, degrading embedding quality and RAG retrieval relevance.

External solutions like Docling provide hierarchical/hybrid chunking but require a separate service and add infrastructure complexity.

**Context**
Semantic chunking is a well-established technique in RAG pipelines, available natively in LangChain (Python) and LangChain4j (Java), but not yet in Spring AI.

Users currently have to either rely on external tools (Docling) or implement custom solutions. A native SemanticTextSplitter extending the existing TextSplitter base class would require no new external dependencies since it builds on Spring AI's own EmbeddingModel interface.

Configurable parameters would include: similarity threshold, max chunk size, and sentence splitting strategy.

### PR DESCRIPTION
**Pull request by anuragg-saxenaa**
## Summary

Add `SemanticTextSplitter` — a new `TextSplitter` that uses semantic similarity rather than fixed token counts to determine chunk boundaries.

Closes #5464

## Motivation

`TokenTextSplitter` splits at fixed token counts, often breaking sentences mid-thought and degrading RAG retrieval quality. Users frequently reach for external tools (e.g., Docling) or write custom solutions. A native semantic splitter addresses this gap without introducing new external dependencies.

## Implementation

`SemanticTextSplitter` extends the existing `TextSplitter` base class:

1. **Sentence tokenisation** — split on `.`, `!`, `?` followed by whitespace.
2. **Embedding** — embed each sentence via the injected `EmbeddingModel`.
3. **Cosine similarity** — compute similarity between consecutive sentence embeddings.
4. **Chunking** — start a new chunk when similarity drops below `similarityThreshold` _or_ the buffer would exceed `maxChunkSize` characters.

**Defaults:** `similarityThreshold = 0.5`, `maxChunkSize = 1000`.
**No new dependencies** — reuses Spring AI's own `EmbeddingModel`.

### Example

```java
SemanticTextSplitter splitter = SemanticTextSplitter.builder()
    .embeddingModel(embeddingModel)
    .similarityThreshold(0.6)
    .maxChunkSize(800)
    .build();

List<Document> chunks = splitter.split(document);
```

## Tests (`SemanticTextSplitterTests`)

| Test | What it verifies |
|------|-----------------|
| `emptyTextReturnsEmptyList` | blank/empty input |
| `singleSentenceReturnsSingleChunk` | single-sentence edge case |
| `highSimilarityKeepsSentencesTogether` | identical embeddings → one chunk |
| `lowSimilaritySplitsSentences` | orthogonal embeddings → separate chunks |
| `maxChunkSizeForcesNewChunk` | size cap overrides similarity |
| `thresholdExactlyAtSimilarityKeepsTogether` | boundary: `sim >= threshold` |
| `thresholdJustAboveSimilaritySplits` | boundary: `sim < threshold` |
| `splitDocumentPreservesMetadata` | metadata propagation |
| `cosineSimilarity*` (×3) | math helper correctness |
| Constructor validation (×4) | null model, invalid threshold/size |

### CODE REVIEW

**Comment by anuragg-saxenaa** [comment]
Closing this PR. Apologies for the noise — going forward all contributions will go through proper human review before submission.

File: spring-ai-commons/src/main/java/org/springframework/ai/transformer/splitter/SemanticTextSplitter.java [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+
+	/**
+	 * Create a {@code SemanticTextSplitter} with explicit parameters.
+	 * @param embeddingModel the embedding model used to compute sentence embeddings
+	 * @param similarityThreshold cosine-similarity threshold in {@code [0, 1]}; a new
+	 * chunk is started when similarity falls below this value
+	 * @param maxChunkSize maximum character length of a single chunk; sentences exceeding
+	 * this on their own are kept as individual chunks regardless of similarity
+	 */
+	public SemanticTextSplitter(EmbeddingModel embeddingModel, double similarityThreshold, int maxChunkSize) {
```
Comment (charrypi) [inline spring-ai-commons/src/main/java/org/springframework/ai/transformer/splitter/SemanticTextSplitter.java:121]: 希望再开放一个构造器，增加SENTENCE_SPLIT_REGEX变量，因为目前的设置对中文不友好，中文的分句不按这种方式。
